pkg/internal/diskutil.go

```diff
@@ -120,6 +122,27 @@ func (b BlockDevice) HasChildren() (bool, error) {
 	return false, nil
 }
 
+// HasBindMounts checks for bind mounts and returns mount point for a device by parsing `proc/1/mountinfo`.
+// HostPID should be set to true inside the POD spec to get details of host's mount points inside `proc/1/mountinfo`.
+func (b BlockDevice) HasBindMounts() (bool, string, error) {
+	data, err := ioutil.ReadFile(mountFile)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to read file %s: %v", mountFile, err)
+	}
+
+	mountString := string(data)
+	for _, mountInfo := range strings.Split(mountString, "\n") {
+		if strings.Contains(mountInfo, b.Name) {
+			mountInfoList := strings.Split(mountInfo, " ")
+			if len(mountInfoList) > 5 && mountInfoList[3] == fmt.Sprintf("/%s", b.Name) {
```

Comment:
For bind mounts the 4th field is the device source but for regular mounts the device source will be present in 10th field and hence we should check for match with either 4th or 10th field.

Reply:
Thanks. Updated.